fgpyo/fastx/__init__.py

```diff
@@ -38,6 +38,8 @@
 from pysam import FastxFile
 from pysam import FastxRecord
 
+from fgpyo.util.types import all_not_none
```

Comment:
I tried to address the type errors with minimal changes to the code, but here adding a helper to narrow the type of a sequence of `Optional[T]` to `T` seemed like the least bad of the available options.